ledger/metrics/src/main/scala/com/daml/metrics/api/opentelemetry/OpenTelemetryFactory.scala

```diff
@@ -141,39 +159,64 @@ case class OpentelemetryTimer(name: String, histogram: LongHistogram) extends Ti
 }
 
 case class OpentelemetryGauge[T](name: String, varGauge: VarGauge[T]) extends Gauge[T] {
-  override def updateValue(newValue: T)(implicit
-      context: MetricsContext
-  ): Unit = varGauge.updateValue(newValue)
+  override def updateValue(newValue: T): Unit = varGauge.updateValue(newValue)
   override def getValue: T = varGauge.getValue
 }
 
-case class OpentelemetryMeter(name: String, counter: LongCounter) extends Meter {
+case class OpentelemetryMeter(name: String, counter: LongCounter, meterContext: MetricsContext)
+    extends Meter {
   override def mark(value: Long)(implicit
       context: MetricsContext
-  ): Unit = counter.add(value, context.asAttributes)
+  ): Unit = counter.add(value, AttributesHelper.multiContextAsAttributes(meterContext, context))
 }
 
-case class OpentelemetryCounter(name: String, counter: LongUpDownCounter) extends Counter {
+case class OpentelemetryCounter(
+    name: String,
+    counter: LongUpDownCounter,
+    counterContext: MetricsContext,
+) extends Counter {
   override def inc()(implicit
       context: MetricsContext
-  ): Unit = counter.add(1, context.asAttributes)
+  ): Unit = counter.add(1, AttributesHelper.multiContextAsAttributes(counterContext, context))
   override def inc(n: Long)(implicit
       context: MetricsContext
-  ): Unit = counter.add(n, context.asAttributes)
+  ): Unit = counter.add(n, AttributesHelper.multiContextAsAttributes(counterContext, context))
   override def dec()(implicit
       context: MetricsContext
-  ): Unit = counter.add(-1, context.asAttributes)
+  ): Unit = counter.add(-1, AttributesHelper.multiContextAsAttributes(counterContext, context))
   override def dec(n: Long)(implicit
       context: MetricsContext
-  ): Unit = counter.add(-n, context.asAttributes)
+  ): Unit = counter.add(-n, AttributesHelper.multiContextAsAttributes(counterContext, context))
   override def getCount: Long = 0 // Not supported by OpenTelemetry
 }
 
-case class OpentelemetryHistogram(name: String, histogram: LongHistogram) extends Histogram {
+case class OpentelemetryHistogram(
+    name: String,
+    histogram: LongHistogram,
+    histogramContext: MetricsContext,
+) extends Histogram {
   override def update(value: Long)(implicit
       context: MetricsContext
-  ): Unit = histogram.record(value, context.asAttributes)
+  ): Unit =
+    histogram.record(value, AttributesHelper.multiContextAsAttributes(histogramContext, context))
   override def update(value: Int)(implicit
       context: MetricsContext
-  ): Unit = histogram.record(value.toLong, context.asAttributes)
+  ): Unit = histogram.record(
+    value.toLong,
+    AttributesHelper.multiContextAsAttributes(histogramContext, context),
+  )
+}
+
+private object AttributesHelper {
+
+  def multiContextAsAttributes(context: MetricsContext*): Attributes = {
```

Comment:
Can we elaborate the expected behaviour for attributes that are defined in multiple contexts, please?